### PR TITLE
[ICDS Dashboard]: Child Growth Tracker Report

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
[Child Growth Tracker Report__27-04-2020__11-47-30.xlsx](https://github.com/dimagi/commcare-hq/files/4539381/Child.Growth.Tracker.Report__27-04-2020__11-47-30.xlsx)
[Child Growth Tracker Report__27-04-2020__11-49-04.zip](https://github.com/dimagi/commcare-hq/files/4539382/Child.Growth.Tracker.Report__27-04-2020__11-49-04.zip)

##### SUMMARY
Child Growth Tracker Report
https://app.asana.com/0/1112385193248823/1165995517877377/f

##### FEATURE FLAG
Yes

##### PRODUCT DESCRIPTION

COST PLAN (Fetching one month data) - District level
```
                                                                                                      QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.block_name, remote_scan.supervisor_name, remote_scan.awc_name, remote_scan.person_name
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Nested Loop  (cost=1288.26..49346.71 rows=7628 width=523)
                     ->  Seq Scan on icds_months_102841 months  (cost=0.00..1.52 rows=1 width=4)
                           Filter: (start_date = '2020-04-01'::date)
                     ->  Gather  (cost=1288.26..48467.96 rows=7628 width=455)
                           Workers Planned: 2
                           ->  Nested Loop  (cost=288.26..46705.16 rows=3178 width=455)
                                 ->  Parallel Bitmap Heap Scan on awc_location_102840 awc_location  (cost=287.83..5631.34 rows=2139 width=241)
                                       Recheck Cond: (district_id = 'b0628d2fdad64375ba41d54280803a95'::text)
                                       ->  Bitmap Index Scan on awc_location_indx3_102840  (cost=0.00..286.55 rows=5133 width=0)
                                             Index Cond: (district_id = 'b0628d2fdad64375ba41d54280803a95'::text)
                                 ->  Append  (cost=0.42..18.26 rows=94 width=247)
                                       ->  Index Scan using "child_health_monthly_2020-04-01_616572_awc_id_idx" on "child_health_monthly_2020-04-01_616572" child_health_monthly  (cost=0.42..17.79 rows=94 width=247)
                                             Index Cond: (awc_id = awc_location.doc_id)
                                             Filter: (month = '2020-04-01'::date)
```
COST PLAN (Fetching three months data) - DISTRICT LEVEL
```
  QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.block_name, remote_scan.supervisor_name, remote_scan.awc_name, remote_scan.person_name
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Gather  (cost=1002.59..446919.73 rows=1488 width=520)
                     Workers Planned: 4
                     ->  Hash Join  (cost=2.59..445770.93 rows=372 width=520)
                           Hash Cond: (child_health_monthly.month = months.start_date)
                           ->  Nested Loop  (cost=0.98..445633.54 rows=24798 width=452)
                                 ->  Parallel Index Scan using awc_location_indx6_102840 on awc_location_102840 awc_location  (cost=0.55..107407.25 rows=1283 width=241)
                                       Filter: (district_id = 'b0628d2fdad64375ba41d54280803a95'::text)
                                 ->  Append  (cost=0.42..249.80 rows=1382 width=243)
                                       ->  Index Scan using "child_health_monthly_2020-01-01_403772_awc_id_idx" on "child_health_monthly_2020-01-01_403772" child_health_monthly  (cost=0.42..15.67 rows=91 width=245)
                                             Index Cond: (awc_id = awc_location.doc_id)
                                       ->  Index Scan using "child_health_monthly_2020-02-01_512124_awc_id_idx" on "child_health_monthly_2020-02-01_512124" child_health_monthly_1  (cost=0.42..16.23 rows=93 width=245)
                                             Index Cond: (awc_id = awc_location.doc_id)
                                       ->  Index Scan using "child_health_monthly_2020-03-01_605564_awc_id_idx" on "child_health_monthly_2020-03-01_605564" child_health_monthly_2  (cost=0.42..17.53 rows=94 width=245)
                                             Index Cond: (awc_id = awc_location.doc_id)
                                       ->  Index Scan using "child_health_monthly_2020-04-01_616572_awc_id_idx" on "child_health_monthly_2020-04-01_616572" child_health_monthly_3  (cost=0.42..17.56 rows=94 width=247)
                                             Index Cond: (awc_id = awc_location.doc_id)
                                       ->  Index Scan using child_health_monthly_default_102648_awc_id_idx on child_health_monthly_default_102648 child_health_monthly_4  (cost=0.56..175.91 rows=1010 width=242)
                                             Index Cond: (awc_id = awc_location.doc_id)
                           ->  Hash  (cost=1.58..1.58 rows=3 width=4)
                                 ->  Seq Scan on icds_months_102841 months  (cost=0.00..1.58 rows=3 width=4)
                                       Filter: (start_date = ANY ('{2020-02-01,2020-03-01,2020-04-01}'::date[]))
```

COST PLAN (Fetching one month data) - AWC LEVEL
```
 QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.person_name
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Nested Loop  (cost=0.98..113.24 rows=86 width=523)
                     ->  Seq Scan on icds_months_102841 months  (cost=0.00..1.52 rows=1 width=4)
                           Filter: (start_date = '2020-04-01'::date)
                     ->  Nested Loop  (cost=0.98..101.83 rows=86 width=455)
                           ->  Index Scan using awc_location_indx6_102840 on awc_location_102840 awc_location  (cost=0.55..2.77 rows=1 width=241)
                                 Index Cond: (doc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                           ->  Append  (cost=0.42..98.20 rows=86 width=247)
                                 ->  Index Scan using "child_health_monthly_2020-04-01_611068_awc_id_idx" on "child_health_monthly_2020-04-01_611068" child_health_monthly  (cost=0.42..97.77 rows=86 width=247)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                                       Filter: (month = '2020-04-01'::date)
```
COST PLAN (Fetching all the three months data) - AWC LEVEL 

```
                                                                                                    QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.person_name
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Nested Loop  (cost=2.59..1259.27 rows=16 width=520)
                     ->  Index Scan using awc_location_indx6_102840 on awc_location_102840 awc_location  (cost=0.55..2.77 rows=1 width=241)
                           Index Cond: (doc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                     ->  Hash Join  (cost=2.04..1254.67 rows=16 width=244)
                           Hash Cond: (child_health_monthly.month = months.start_date)
                           ->  Append  (cost=0.42..1248.78 rows=1097 width=243)
                                 ->  Index Scan using "child_health_monthly_2020-01-01_403772_awc_id_idx" on "child_health_monthly_2020-01-01_403772" child_health_monthly  (cost=0.42..94.21 rows=83 width=245)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                                 ->  Index Scan using "child_health_monthly_2020-02-01_512124_awc_id_idx" on "child_health_monthly_2020-02-01_512124" child_health_monthly_1  (cost=0.42..96.47 rows=85 width=245)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                                 ->  Index Scan using "child_health_monthly_2020-03-01_605564_awc_id_idx" on "child_health_monthly_2020-03-01_605564" child_health_monthly_2  (cost=0.42..97.55 rows=86 width=245)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                                 ->  Index Scan using "child_health_monthly_2020-04-01_611068_awc_id_idx" on "child_health_monthly_2020-04-01_611068" child_health_monthly_3  (cost=0.42..97.56 rows=86 width=247)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                                 ->  Index Scan using child_health_monthly_default_102648_awc_id_idx on child_health_monthly_default_102648 child_health_monthly_4  (cost=0.56..857.51 rows=757 width=242)
                                       Index Cond: (awc_id = '02a5f8f0e3e04dd5afca21bd58681b56'::text)
                           ->  Hash  (cost=1.58..1.58 rows=3 width=4)
                                 ->  Seq Scan on icds_months_102841 months  (cost=0.00..1.58 rows=3 width=4)
                                       Filter: (start_date = ANY ('{2020-02-01,2020-03-01,2020-04-01}'::date[]))
```

### CODE REVIEW

**Comment by czue** [comment]
(not planning on reviewing this, let me know if you need me to)

File: custom/icds_reports/sqldata/exports/growth.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+from corehq.apps.locations.models import SQLLocation
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth.py:1]: File name could be more specific. may be something like `growth_tracker_report.py`

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 2]
```diff
@@ -507,6 +509,10 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         return vm.selectedIndicator === 11;
     };
 
+    vm.isChildGrowthSelected = function () {
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/download/download.directive.js:512]: can we call it as growth tracker? So that no one confuses it with growth monitoring list

File: custom/icds_reports/migrations/sql_templates/database_views/child_health_monthly.sql [hunk 3]
```diff
@@ -10,7 +10,9 @@ CREATE VIEW child_health_monthly_view AS
       awc_location_months.block_id,
       awc_location_months.block_name,
       awc_location_months.district_id,
+      awc_location_months.district_name,
```
Comment (Rohit25negi) [inline custom/icds_reports/migrations/sql_templates/database_views/child_health_monthly.sql:13]: you will also need to add the migration file for ths.

File: custom/icds_reports/sqldata/exports/growth.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    table_name = 'child_health_monthly_view'
+
+    def get_excel_data(self, location):
+
+        def _check_case_presence(case_id, column, data_dict):
+            return data_dict[case_id][column] if case_id in data_dict.keys() else "N/A"
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth.py:22]: does this spec ask to put this limit explicitly because indicators in this report, PSE and LUNCH are given from 3 to 6 years.

File: custom/icds_reports/sqldata/exports/growth.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name')
+        elif self.loc_level == 1:
+            filters['state_id'] = location
+            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name')
+        else:
+            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
+
+        query_set_1 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth.py:45]: these three calls to the database table is causing code repetition can be put this in a function and call is thrice instead?

File: custom/icds_reports/sqldata/exports/growth.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        filters["month"] = initial_month - relativedelta(months=1)
+        query_set_2 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+        data_past_month = query_set_2.values('case_id', 'pse_days_attended', 'lunch_count',
+                                             'current_month_nutrition_status',
+                                             current_month_wasting_column(self.beta),
+                                             current_month_stunting_column(self.beta), 'case_id')
+        data_past_month = {item['case_id']: item for item in data_past_month}
+
+        filters["month"] = initial_month - relativedelta(months=2)
+        query_set_3 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth.py:62]: For weighing the performance. Could you put the cost of following here:
1. If you pull all the three months data together
2. Doing it separately like this.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left a few broader comments. I also left a question in the spec. I think that spec is not right for the UI

File: custom/icds_reports/sqldata/exports/growth.py (hunk 4 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth.py:22]: for Both PSE and Lunch there is a lower bound (not sure if there is a upper bound or not) that's why keep this

File: custom/icds_reports/sqldata/exports/growth.py (hunk 5 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth.py]: updated!!

File: custom/icds_reports/sqldata/exports/growth.py (hunk 6 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth.py]: Added!

File: custom/icds_reports/sqldata/exports/growth.py (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth.py]: updated!

File: custom/icds_reports/static/js/directives/download/download.directive.js (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: updated!!

**Comment by ShubhamBansal1997** [comment]
@Rohit25negi Updated the PR

File: custom/icds_reports/migrations/sql_templates/database_views/child_health_monthly.sql (hunk 3 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/migrations/sql_templates/database_views/child_health_monthly.sql:13]: added

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name','person_name')
+        elif self.loc_level == 1:
+            filters['state_id'] = location
+            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name','person_name')
+        else:
+            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name','person_name')
+
+        # Sample cost of each query(if data fetched for single month) is 0.98..3012.55
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:57]: hey, no need to put the comments here like this. I think putting it in Pr comments should be enough. Also, single cost count wont be enough. Would be great if you could paste the entire plan here for both the queries

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        def _fetch_data(filters, order_by, case_by_grouping=False):
+            query_set = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+            data_month = query_set.values('state_name', 'district_name', 'block_name', 'supervisor_name',
+                                          'awc_name', 'awc_site_code', 'person_name', 'dob', 'mother_name',
+                                          'mother_phone_number', 'pse_days_attended', 'lunch_count',
+                                          'current_month_nutrition_status',
+                                          current_month_wasting_column(self.beta),
+                                          current_month_stunting_column(self.beta), 'case_id')
+            if case_by_grouping is True:
+                data_month = {item['case_id']: item for item in data_month}
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:29]: Why is this required?

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+        data_initial_month = _fetch_data(filters, order_by)
+        filters["month"] = initial_month - relativedelta(months=1)
+        data_past_month = _fetch_data(filters, order_by, True)
+        filters["month"] = initial_month - relativedelta(months=2)
+        data_past_month_2 = _fetch_data(filters, order_by, True)
+        filters["month"] = initial_month
+
+
+        month_1 = (initial_month - relativedelta(months=2)).strftime('%Y-%m-%d')
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:69]: good to have the convention the same with the above  data variables above. You could change any of the ones to match to another.

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+                    True
+                )
+            ]
+            excel_rows.append(row_data)
+        filters = [['Generated at', india_now()]]
+        if location:
+            locs = SQLLocation.objects.get(location_id=location).get_ancestors(include_self=True)
+            for loc in locs:
+                filters.append([loc.location_type.name.title(), loc.name])
+        else:
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:162]: Is this report available at National level?

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+                                          'current_month_nutrition_status',
+                                          current_month_wasting_column(self.beta),
+                                          current_month_stunting_column(self.beta), 'case_id')
+            if case_by_grouping is True:
+                data_month = {item['case_id']: item for item in data_month}
+            return data_month
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:34]: hey again, Why is there an upper bound of 5 years here. because PSE and SNP are given until 6 years?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left a bunch of comments. 
Also have you test this locally? Can you upload the csv and xlsx files generated locally. Good to also test this that it is under feature flag

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 8 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:29]: We are creating dict for past months we used to pick the case_id in O(1) time while structuring data

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 10 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:162]: No

File: custom/icds_reports/migrations/0183_auto_20200420_1038.py [hunk 12]
```diff
@@ -20,5 +20,3 @@ class Migration(migrations.Migration):
         migrations.RunSQL("ALTER TABLE child_health_monthly ADD COLUMN delivery_nature text"),
         migrations.RunSQL("ALTER TABLE child_health_monthly ADD COLUMN term_days smallint"),
     ]
-
-    operations.extend(get_view_migrations())
```
[the remark this answers is not in the document]
Comment (ShubhamBansal1997) [reply custom/icds_reports/migrations/0183_auto_20200420_1038.py:24]: It's mentioned by challa to rebuild views on the latest migration

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+        elif self.loc_level == 4:
+            filters['supervisor_id'] = location
+            order_by = ('awc_name','person_name')
+        elif self.loc_level == 3:
+            filters['block_id'] = location
+            order_by = ('supervisor_name', 'awc_name','person_name')
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name','person_name')
+        elif self.loc_level == 1:
```
[the remark this answers is not in the document]
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:51]: removing

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 11 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py]: updated

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 9 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:59]: updated

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 7 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py]: updated

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+                                          'current_month_nutrition_status',
+                                          current_month_wasting_column(self.beta),
+                                          current_month_stunting_column(self.beta), 'case_id')
+            if case_by_grouping is True:
+                data_month = {item['case_id']: item for item in data_month}
+            return data_month
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 72,
```
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/growth_tracker_report.py:35]: Can you confirm what do we want to fill in cell for the indicator which do not apply for  the children. Like stunting does not apply for 5-6 years of children.

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 9 above)
Comment (Rohit25negi) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:59]: THanks

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Overall looks good. Left few small comments. Also the query plan you pasted which level you check that plan? At awc level? can you put the plan of running at district level?

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 14 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:35]: Talked this regarding the GS team, they said let's keep it as same as dashboard

File: custom/icds_reports/sqldata/exports/growth.py (hunk 4 above)
Comment (nitishdubedimagi) [reply custom/icds_reports/sqldata/exports/growth.py]: Upper bound should not be 60. This report would include Children 5-6 yrs as well

File: custom/icds_reports/sqldata/exports/growth_tracker_report.py (hunk 14 above)
Comment (nitishdubedimagi) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:35]: Will this also take care of the case where a past month's data is not available?

Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth_tracker_report.py:35]: It will automatically fill N/A for all of them

File: custom/icds_reports/sqldata/exports/growth.py (hunk 4 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/sqldata/exports/growth.py]: It's outdated, it's updated to 72

File: custom/icds_reports/migrations/0183_auto_20200420_1038.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (ShubhamBansal1997) [reply custom/icds_reports/migrations/0183_auto_20200420_1038.py]: updated

**Review by Rohit25negi — APPROVED** [review]
Looks good to me. The report may be slow on district level. But I am not sure how much, we will have to see if the performance is good enough to be accepted once that is deployed.  
I have also kicked the build again to ensure that it's not broken because of other conflicting changes being merged
